src/videojs-contrib-hls.js

```diff
@@ -273,6 +276,11 @@ export default class HlsHandler extends Component {
 const HlsSourceHandler = function(mode) {
   return {
     canHandleSource(srcObj) {
+      // this forces video.js to skip this tech/mode if its not the one we have been
+      // overriden to use, by returing that we cannot handle the source.
+      if (typeof videojs.options.hls.mode === 'string' && mode !== videojs.options.hls.mode) {
```

Comment:
> This would restrict when the mode can be overridden.

I don't understand what "this" is. My point was you may be dereferencing `undefined` in this line.